### PR TITLE
New UX for storage selector

### PR DESCRIPTION
**Pull request by donnapep**

### CODE REVIEW

**Comment by donnapep** [comment]
@rodrigopavezi @stulees Please review. Thx,

**Comment by stulees** [comment]
@donnapep Can you clarify how the component doesn't require any change in URL Field here - https://github.com/Rise-Vision/widget-settings-ui-components/blob/master/src/_angular/url-field/url-field.html#L5? It requires it to be the icon only and to allow for setting its selector type (`file` or `folder`). Will these changes break this implementation?

**Comment by donnapep** [comment]
@stulees Rodrigo is making the required changes to the URL field component. As part of those changes, storage selector will be removed from it.

**Comment by stulees** [comment]
@donnapep @rodrigopavezi Okay, thanks for clarifying. I assume then Background Image Setting will also require updating as it uses the current URL Field with a storage selector?

**Comment by rodrigopavezi** [comment]
@stulees that is right, we will have to it. cheers

File: bower.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "component-storage-selector",
-  "version": "1.0.1",
+  "version": "1.0.2",
```
Comment (stulees) [inline bower.json]: @donnapep These changes look to be a minor change that involve a new feature, version should be `1.1.0`.

**Comment by stulees** [comment]
@donnapep @rodrigopavezi Sorry, still confused. Why are all three buttons always showing? How do you use this component to just show one icon button? My interpretation from the Settings UX was that it would be 3 different instances of this component, each one set to display the appropriate visual and selector type. Or am I missing something?

**Comment by rodrigopavezi** [comment]
@stulees I got what you are saying. Background image settings does not work with folders so I suggest we still keep on storage-selector with the option to show/hide the buttons. For instance, if folder-label is not on the element it should not show the select folder button? What do you guys think? @donnapep FYI.

**Comment by stulees** [comment]
@rodrigopavezi That won't work when wanting to show the button with no label but have it be for a single storage file (Background Image Setting requires it this way).

The implementation I saw towards this component was mainly a visual one. I thought `type` would still persist and take a third value of `custom`, and based on `type` it would determine which button UX to show. A new attribute could be added like `icon-only` that would ensure only the icon shows if set to true. This way the component maintains its flexibility in being used in different contexts.

**Comment by donnapep** [comment]
@stulees Thanks. I actually bumped the version to 2.0.0 since it's not backwards compatible.

@stulees @rodrigopavezi I don't think we need to worry about background image setting right now. It is not part of the work we are currently doing for the Video Widget epic. We would probably have to address this as part of the Image Widget epic. As such, I don't see a need to implement showing/hiding now.

Having said that, is the way it currently works going to cause a problem further down the line? If yes, then we should change it. If not, then let's enhance the existing functionality as required (if it's ever required) later on.

Thx.

**Pull request closed without merging**